cfme/tests/physical_infrastructure/ui/test_physical_server_list_buttons.py

```diff
@@ -0,0 +1,109 @@
+# -*- coding: utf-8 -*-
+import pytest
+from cfme.physical.provider.lenovo import LenovoProvider
+from cfme.utils.appliance.implementations.ui import navigate_to
+from cfme.common.physical_server_views import PhysicalServersView
+from cfme.utils.wait import wait_for
+
+pytestmark = [pytest.mark.tier(3), pytest.mark.provider([LenovoProvider], scope="module")]
+
+
+@pytest.fixture(scope="function")
+def physical_servers(appliance, provider, setup_provider):
+    # Get and return all the physical servers
+    return appliance.collections.physical_servers.all(provider)
+
+
+@pytest.fixture(scope="function")
+def physical_servers_collection(appliance):
+    # Get and return collection of the physical servers
+    return appliance.collections.physical_servers
+
+
+# Configuration Button
+def test_refresh_relationships(physical_servers_collection, physical_servers, provider):
+    view = provider.create_view(PhysicalServersView)
+    last_refresh = provider.last_refresh_date()
+    option = "Refresh Relationships and Power States"
+    physical_servers_collection.custom_button_action("Configuration", option, physical_servers)
+    wait_for(
+        lambda: last_refresh != provider.last_refresh_date(),
+        fail_func=view.browser.refresh,
+        message="Wait for the servers to be refreshed...",
+        num_sec=300,
+        delay=20
+    )
+    assert last_refresh != provider.last_refresh_date()
+
+
+actions = [
+    ("Power", "Power Off", "power_off"),
+    ("Power", "Power On", "power_on"),
+    ("Power", "Power Off Immediately", "power_off_now"),
+    ("Power", "Restart", "restart"),
+    ("Power", "Restart Immediately", "restart_now"),
+    ("Power", "Restart to System Setup", "restart_to_sys_setup"),
+    ("Power", "Restart Management Controller", "restart_mgmt_controller"),
+    ("Identify", "Blink LED", "blink_loc_led"),
+    ("Identify", "Turn Off LED", "turn_off_loc_led"),
+    ("Identify", "Turn On LED", "turn_on_loc_led")
+]
+
+
+# Power / Identify Buttons
+@pytest.mark.parametrize("button, option, method_name",
+                         actions, ids=[action[1] for action in actions])
+def test_server_actions(physical_servers_collection, physical_servers, provider,
+                        button, option, method_name):
+    """ Test the physical server actions are creating a handler alert to each action of the a collection
+    of physical servers.
+    Params:
+        * button: the button to be performed on the physical server list page
+        * option: the option to be selected inside the dropdrown button
+        * method_name: the name of the method that most be used to compare if was invoked the
+        current method on the manageIQ.
+    Metadata:
+        test_flag: crud
+    """
+    view = provider.create_view(PhysicalServersView)
+    last_part = ""
```

Comment:
**Optional**

express this as ternary, `last_part = 's' if len(physical_servers) > 1 else ''`

Reply:
I like this, I will to use your suggestion.

Reply:
done